addons/product/models/product_template.py

```diff
@@ -428,7 +429,7 @@ def create_variant_ids(self):
 
             # create new product
             for variant_ids in to_create_variants:
-                new_variant = Product.create({
+                new_products |= Product.create({
```

Comment:
That's unnecessary, `recompute()` just triggers an execution of all pending stored function field computations, it doesn't care about the recorset (it doesn't even care about the model it's called on, that's just an API convenience).